src/components/RouteForDeliveryModal/RouteForDeliveryModal.test.js

```diff
@@ -0,0 +1,171 @@
+import React from 'react';
+import {
+  render,
+  screen,
+  fireEvent,
+} from '@testing-library/react';
+
+import '../../../test/jest/__mock__';
+
+import {
+  Modal,
+  Button,
+  Checkbox,
+} from '@folio/stripes/components';
+
+import RouteForDeliveryModal from './RouteForDeliveryModal';
+import PrintButton from '../PrintButton';
+
+jest.mock('../PrintButton', () => jest.fn(({ children, ...rest }) => (
+  <button type="button" data-testid="print-button" {...rest}>
+    {children}
+  </button>
+)));
+
+const defaultProps = {
+  label: 'test-label',
+  modalContent: 'test-modal-content',
+  open: true,
+  onCloseAndCheckout: jest.fn(),
+  onClose: jest.fn(),
+  slipTemplate: 'test-template',
+  slipData: {},
+  isPrintableByDefault: false,
+};
+
+const messageIds = {
+  printSlip: 'ui-checkin.statusModal.printSlip',
+  close: 'ui-checkin.statusModal.close',
+  closeAndCheckout: 'ui-checkin.statusModal.delivery.closeAndCheckout',
+};
+
+describe('RouteForDeliveryModal', () => {
+  const buttonCallOrder = {
+    closeAndCheckout: 1,
+    close: 2,
+  };
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('Initial render with default props', () => {
+    beforeEach(() => {
+      render(
+        <RouteForDeliveryModal
+          {...defaultProps}
+        />
+      );
+    });
+
+    it('"Modal" component should be called with correct props', () => {
+      const expectedProps = {
+        dismissible: true,
+        onClose: defaultProps.onClose,
+        open: defaultProps.open,
+        label: defaultProps.label,
+      };
+
+      expect(Modal).toHaveBeenCalledWith(expect.objectContaining(expectedProps), {});
+    });
+
+    it('Modal content should be rendered', () => {
+      expect(screen.getByText(defaultProps.modalContent)).toBeInTheDocument();
+    });
+
+    it('"Checkbox" component should be called with correct props', () => {
+      const expectedProps = {
+        name: 'printSlip',
+        onChange: expect.any(Function),
+        checked: defaultProps.isPrintableByDefault,
+        value: defaultProps.isPrintableByDefault,
+      };
+
+      expect(Checkbox).toHaveBeenCalledWith(expect.objectContaining(expectedProps), {});
+    });
+
+    it('Appropriate label of "Checkbox" component should be rendered', () => {
+      expect(screen.getByText(messageIds.printSlip)).toBeInTheDocument();
+    });
+
+    it('"Close and checkout" button should be called with correct props', () => {
+      const expectedProps = {
+        buttonStyle: 'primary',
+        onClick: defaultProps.onCloseAndCheckout,
+      };
+
+      expect(Button).toHaveBeenNthCalledWith(buttonCallOrder.closeAndCheckout, expect.objectContaining(expectedProps), {});
```

Comment:
As we already have [this helper](https://github.com/folio-org/ui-checkin/blob/6b884115f92d3e77c0cd0e6bbe4acfd9b42a186b/test/jest/helpers/utils.js#L8-L21) in current module, maybe will be better to use it here? Instead of call order.
But it's up to you.

Reply:
If you do not mind I will leave it as is. But in the future will try to use a helper.